app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt

```diff
@@ -73,99 +73,217 @@ class StringToFractionParserTest {
+    val error = stringToFractionParser.getRealTimeAnswerError("-5 2/3")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.VALID)
+  }
+
+  @Test
+  fun testParser_realTimeError_returnsInvalidFormat() {
+    val error = stringToFractionParser.getRealTimeAnswerError("/23/")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_realTimeError_returnsInvalidChars() {
+    val error = stringToFractionParser.getRealTimeAnswerError("abc")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_CHARS)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please only use numerical digits, spaces or forward slashes (/)")
   }
 
   @Test
   fun testParser_realTimeError_noNumerator_returnsInvalidFormat() {
     val error = stringToFractionParser.getRealTimeAnswerError("/3")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
   }
 
   @Test
   fun testParser_realTimeError_severalSlashes_returnsInvalidFormat() {
     val error = stringToFractionParser.getRealTimeAnswerError("1/3/8")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
   }
 
   @Test
   fun testParser_realTimeError_severalDashes_returnsInvalidFormat() {
     val error = stringToFractionParser.getRealTimeAnswerError("-1/-3")
-      .getErrorMessageFromStringRes(context)
-    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+    assertThat(error).isEqualTo(StringToFractionParser.FractionParsingError.INVALID_FORMAT)
+
+    val errorMessage = error.getErrorMessageFromStringRes(context)
+    assertThat(errorMessage)
+      .isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_parseFraction_divisionByZero() {
+    val parseFraction = stringToFractionParser.parseFraction("8/0")
+    val parseFractionFromString = stringToFractionParser.parseFractionFromString("8/0")
+    val constructedFraction = Fraction.newBuilder().apply {
+      numerator = 8
```

Comment:
Nit, but I suggest explicitly defining denominator to be 0 (even though it defaults) just to be really explicit about the test's behavior.

Reply:
Done.